src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
I've solved it in this branch: 
https://github.com/silo-co/phonegap-plugin-push
You can read about it here:
http://www.yossale.com/Ionic-android-icon-badges/